### ISSUE TITLE
different atten-mask depending on whether xformer is installed

### ISSUE THREAD

**Issue by chenkq7** [Luodian/Otter#118]
hi, thanks for your awesome work, which helps me a lot.

i'm trying to deploy otter locally. diving into the code, i found the otter behaivor is different depending on whether xformer is installed. precisely, when the xformer is not installed, otter use [openflamingo format attention_mask](https://github.com/Luodian/Otter/blob/5a7fcbc5316fe8bccfe972cdc4e65ffcb543a138/otter/modeling_otter.py#L269), when installed, otter use [default attention_mask](https://github.com/Luodian/Otter/blob/5a7fcbc5316fe8bccfe972cdc4e65ffcb543a138/otter/modeling_otter.py#L317).

i was wondering if this will introduce any performance gap between use xformer or not? and the training process used xformer or not?

**Comment by king159** [comment]
The `xformer` attention implementation should behave exactly the same as the original OpenFlamingo one, hence it could potentially reduce memory usage (or improve the training/inference speed depending on your GPU) but will not impact performance. Additionally, our Otter model was trained with `xformer` enabled.
